assets/app/scripts/services/docker.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("DockerImageGeneratorService", function($filter, ApplicationGenerator, DataService) {
+    var findImage = function(name, projectContext) {
+      var importImage = {
+        kind: "ImageStreamImport",
+        apiVersion: "v1",
+        metadata: {
+          name: "newapp",
```

Comment:
doesn't necessarily mean its right ;-)  if it does cause an error, then arguably name should be optional if import is false (probably easier said than done) cc @deads2k